src/applications/auth/containers/AuthApp.jsx

```diff
@@ -1,82 +1,51 @@
 import React from 'react';
-import { connect } from 'react-redux';
 
 import AlertBox from '@department-of-veterans-affairs/formation/AlertBox';
 import LoadingIndicator from '@department-of-veterans-affairs/formation/LoadingIndicator';
 
-import siteName from '../../../platform/brand-consolidation/site-name';
 import CallHelpDesk from '../../../platform/brand-consolidation/components/CallHelpDesk';
-import recordEvent from '../../../platform/monitoring/record-event';
+import siteName from '../../../platform/brand-consolidation/site-name';
+import { setupProfileSession } from '../../../platform/user/profile/utilities';
 import { apiRequest } from '../../../platform/utilities/api';
 import environment from '../../../platform/utilities/environment';
-import localStorage from '../../../platform/utilities/storage/localStorage';
 
 import facilityLocator from '../../facility-locator/manifest';
 
 export class AuthApp extends React.Component {
   constructor(props) {
     super(props);
-    const { auth, token } = props.location.query;
-    this.state = { error: auth === 'fail' || !token };
-    this.authSettings = {
-      headers: {
-        Authorization: `Token token=${token}`,
-      },
-    };
+    this.state = { error: props.location.query.auth === 'fail' };
   }
 
   componentDidMount() {
-    if (!this.state.error) {
-      this.validateToken();
-    }
+    if (!this.state.error) this.validateSession();
   }
 
-  setToken = () => {
+  handleAuthError = () => {
+    this.setState({ error: true });
+  };
+
+  handleAuthSuccess = payload => {
+    setupProfileSession(payload);
+
+    const parent = window.opener;
+    parent.postMessage('loggedIn', environment.BASE_URL);
```

Comment:
I was gonna ask if there's a way we can listen for a cookie-change event, but looks like we'd have a similar issue with [IE support](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/cookies/onChanged).

Reply:
Oh man, that would've been perfect...